pkg/monitor/cluster/nodes.go

```diff
@@ -0,0 +1,41 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (mon *Monitor) emitNodesMetrics() error {
+	nodes, err := mon.cli.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	mon.emitGauge("cluster.nodes.total", int64(len(nodes.Items)), map[string]string{})
+
+	counters := map[string]int64{}
+	for _, node := range nodes.Items {
+		for _, c := range node.Status.Conditions {
+			// count 'Unknown' status as unhealthy state for each condition. In this way
+			// we can flag issues without creating additional timeseries for each condition.
+			// for NodeReady count a node when the status is False (not ready) or Unknown
+			// for other conditions count when the status is True or Unknown
+			if c.Type == corev1.NodeReady && c.Status != corev1.ConditionTrue {
+				counters["NotReadyNodes"]++
+			} else if c.Type != corev1.NodeReady && c.Status != corev1.ConditionFalse {
+				counters[string(c.Type)+"Nodes"]++
+			}
+		}
+	}
+
+	for label, value := range counters {
+		if value > 0 {
```

Comment:
(I think this is always the case)

Reply:
it makes it easier to understand. I think the concept of not emitting metrics at regular intervals is not how many people think of metrics. This will at least make it clearer that this is intentional

Reply:
ok, but there are never any keys in counters whose value is 0.  Currently if a key exists in counters, its value must be 1 or greater.

Reply:
this is true. Removed